--- origsrc/rubrica2-2.0.2/libral/Makefile.am	2007-09-13 15:28:35.000000000 -0500
+++ src/rubrica2-2.0.2/libral/Makefile.am	2007-09-16 04:11:05.531250000 -0500
@@ -85,6 +85,8 @@
 
 libral_la_LIBADD = $(LIBRAL_LIBS)
 
-libral_la_LDFLAGS = -version-info $(LIBRAL_CURRENT_VERSION):$(LIBRAL_REVISION_VERSION):$(LIBRAL_COMPAT_VERSION)
+libral_la_LDFLAGS = -version-info $(LIBRAL_CURRENT_VERSION):$(LIBRAL_REVISION_VERSION):$(LIBRAL_COMPAT_VERSION) -no-undefined
+
+libral_la_LIBADD  = $(LIBRAL_LIBS)
 
 libral_la_CFLAGS  = ${LIBRAL_CFLAGS} -Wall -Wmissing-prototypes -g
--- origsrc/rubrica2-2.0.2/libral/manager.c	2007-09-13 15:28:35.000000000 -0500
+++ src/rubrica2-2.0.2/libral/manager.c	2007-09-16 04:09:24.593750000 -0500
@@ -213,7 +213,7 @@
       g_log(G_LOG_DOMAIN, G_LOG_LEVEL_DEBUG, "Get file: %s", file);
 #endif
 
-      if (g_str_has_suffix(file, ".so"))
+      if (g_str_has_suffix(file, "."G_MODULE_SUFFIX))
 	{
 #ifdef ENABLE_DEBUG
       g_log(G_LOG_DOMAIN, G_LOG_LEVEL_DEBUG, "Reading plugin: %s", file);
--- origsrc/rubrica2-2.0.2/plugins/csv/Makefile.am	2007-09-13 15:28:37.000000000 -0500
+++ src/rubrica2-2.0.2/plugins/csv/Makefile.am	2007-09-16 04:11:05.531250000 -0500
@@ -32,9 +32,9 @@
 # Plugin sources
 libcsv_la_SOURCES = csv.c	csv.h
 
-libcsv_la_LDFLAGS = -version-info $(LIBCSV_CURRENT_VERSION):$(LIBCSV_REVISION_VERSION):$(LIBCSV_COMPAT_VERSION)
+libcsv_la_LDFLAGS = -module -avoid-version -no-undefined
 
-libcsv_LTLDFLAGS = -module -rpath./
+libcsv_la_LIBADD = $(top_builddir)/libral/libral.la $(LIBRAL_LIBS)
 
 
 # Include paths
@@ -43,7 +43,7 @@
 	-DGETTEXT_PACKAGE=\""@GETTEXT_PACKAGE@"\"	\
 	$(LIBRAL_CFLAGS) 	
 
-AM_CFLAGS = -Wall -fPIC -DPIC
+AM_CFLAGS = -Wall
 
 # EXTRA_DIST = 			\
 #	$(csv_glade_DATA) 	\
--- origsrc/rubrica2-2.0.2/plugins/gmail/Makefile.am	2007-09-13 15:28:37.000000000 -0500
+++ src/rubrica2-2.0.2/plugins/gmail/Makefile.am	2007-09-16 04:12:13.750000000 -0500
@@ -30,9 +30,9 @@
 # Plugin sources
 libcsvgmail_la_SOURCES = gmail_csv.c	gmail_csv.h
 
-libcsvgmail_la_LDFLAGS = -version-info $(LIBCSV_GMAIL_CURRENT_VERSION):$(LIBCSV_GMAIL_REVISION_VERSION):$(LIBCSV_GMAIL_COMPAT_VERSION)
+libcsvgmail_la_LDFLAGS = -module -avoid-version -no-undefined
 
-libcsv_gmail_LTLDFLAGS = -module -rpath./
+libcsvgmail_la_LIBADD = $(top_builddir)/libral/libral.la $(LIBRAL_LIBS)
 
 
 # Include paths
--- origsrc/rubrica2-2.0.2/plugins/rubrica/Makefile.am	2007-09-13 15:28:37.000000000 -0500
+++ src/rubrica2-2.0.2/plugins/rubrica/Makefile.am	2007-09-16 04:11:05.546875000 -0500
@@ -35,9 +35,9 @@
 	load.c		load.h		\
 	save.c		save.h
 
-librubrica_la_LDFLAGS = -version-info $(LIBRUBRICA_CURRENT_VERSION):$(LIBRUBRICA_REVISION_VERSION):$(LIBRUBRICA_COMPAT_VERSION)
+librubrica_la_LDFLAGS = -module -avoid-version -no-undefined
 
-librubrica_LTLDFLAGS = -module -rpath./
+librubrica_la_LIBADD = $(top_builddir)/libral/libral.la $(LIBRAL_LIBS)
 
 
 # Include paths
@@ -49,7 +49,7 @@
 	-DRUBRICA_FILE_FORMAT=\""@RUBRICA_FILE_FORMAT@"\"	\
 	$(LIBRAL_CFLAGS) 
 
-AM_CFLAGS = -Wall -fPIC -DPIC
+AM_CFLAGS = -Wall
 
 #EXTRA_DIST = 			\
 #	$(rubrica_glade_DATA) 	\
--- origsrc/rubrica2-2.0.2/plugins/vcard/Makefile.am	2007-09-13 15:28:36.000000000 -0500
+++ src/rubrica2-2.0.2/plugins/vcard/Makefile.am	2007-09-16 04:10:48.687500000 -0500
@@ -32,9 +32,9 @@
 # Plugin sources
 libvcard_la_SOURCES = vcard.c	vcard.h
 
-libvcard_la_LDFLAGS = -version-info $(LIBVCARD_CURRENT_VERSION):$(LIBVCARD_REVISION_VERSION):$(LIBVCARD_COMPAT_VERSION)
+libvcard_la_LDFLAGS = -module -avoid-version -no-undefined
 
-libvcard_LTLDFLAGS = -module -rpath./
+libvcard_la_LIBADD = $(top_builddir)/libral/libral.la $(LIBRAL_LIBS)
 
 
 # Include paths
@@ -43,7 +43,7 @@
 	-DGETTEXT_PACKAGE=\""@GETTEXT_PACKAGE@"\"	\
 	$(LIBRAL_CFLAGS) 
 
-AM_CFLAGS = -Wall -fPIC -DPIC
+AM_CFLAGS = -Wall
 
 #EXTRA_DIST = 			\
 #	$(vcard_glade_DATA) 	\
